integration/update_test.go

```diff
@@ -628,6 +629,46 @@ func TestUpdateMany(t *testing.T) {
 				Message: "Updating the path 'foo' would create a conflict at 'foo'",
 			},
 		},
+		"UnsetString": {
+			filter: bson.D{{"_id", "string"}},
+			update: bson.D{{"$unset", bson.D{{"value", int32(1)}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			res: bson.D{{"_id", "string"}},
+		},
+		"UnsetEmpty": {
+			filter: bson.D{{"_id", "string"}},
+			update: bson.D{{"$unset", bson.D{}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 0,
+				UpsertedCount: 0,
+			},
+			res: bson.D{{"_id", "string"}, {"value", "foo"}},
+		},
+		"UnsetField": {
+			filter: bson.D{{"_id", "document-composite"}},
+			update: bson.D{{"$unset", bson.D{{"value", bson.D{{"array", int32(1)}}}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  0,
+				ModifiedCount: 0,
+				UpsertedCount: 1,
+			},
+			res: bson.D{{"_id", "document-composite"}},
+		},
+		"UnsetEmptyArray": {
+			filter: bson.D{{"_id", "document-composite"}},
+			update: bson.D{{"$unset", bson.A{}}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type array instead. " +
```

Comment:
Why? I see that `$unset` can be an array.

Reply:
> Why? I see that `$unset` can be an array.

That error on an empty array gives MongoDB. To specify an array, dot notation should be used. Dot notation is not in scope.

Reply:
>  Should we give "not implemented" for queries that use dot notation? Or the check is not in scope either?

It should and it's in the other PR by @w84thesun 
https://github.com/FerretDB/FerretDB/pull/660/files#diff-d5aa988135248001431214e8620db5b7b3b57b75f6b692216bc820a1cddb68a5R198

> $unset: ["a", "b", "c"] 


I've already had a test for an empty array `UnsetEmptyArray`. 
This is same check, the error is 
`Modifiers operate on fields but we found type array instead. For example: {$mod: {<field>: ...}} not {$unset: [ \"a\", \"b\", \"c\" ]}`